runtime/src/lib.rs

```diff
@@ -67,6 +67,8 @@ pub enum Error {
     PartitionMissing(String),
     #[error("partition corrupt: {0}")]
     PartitionCorrupt(String),
+    #[error("lock grab failed")]
+    LockGrabFailed,
```

Comment:
Wasn't really sure what error to return in case this happens, so I made a new one. I elected not to include any data in this error type since I expect we'll see it quite rarely, and if we do, something is probably pretty wrong.

As an alternative, I could use, for example `Error::BlobOpenFailed` is this occurs in `open`, or `Error::BlobTruncateFailed` if it happens in `truncate`, and so on.